File: goimp/components/mazes.go
package components

import "fmt"

type Maze struct {
	rooms map[int]Room
}

func createMaze() *Maze {
	return &Maze{
		rooms: make(map[int]Room, 0),
	}
}

func (m *Maze) AddRoom(r Room) {
	m.rooms[r.GetNumber()] = r
}

func (m *Maze) RoomNo(n int) Room {
	return m.rooms[n]
}

func (m *Maze) String() string {
	s := "\nmaze: \n"
	for k, v := range m.rooms {
		s += fmt.Sprintf("%d: %s\n", k, v)
	}
	return s
}

File: py/patterns/components/mazes.py
from .rooms import Room


class Maze:
    def __init__(self) -> None:
        self.__rooms: dict[int, Room] = {}

    def add_room(self, r: Room):
        self.__rooms[r.get_number()] = r

    def get_room(self, n: int) -> any:
        if n in self.__rooms:
            return self.__rooms[n]

    def __str__(self):
        s = 'maze:\n'
        for k in self.__rooms:
            s = "{} {}: {}\n".format(s, k, self.__rooms[k])

        return s
